lib/gameboard/coordinate.rb

```diff
@@ -0,0 +1,60 @@
+module Gameboard
+  class Coordinate
+    # Public: Returns the X,Y coordinate of the instance as [X, Y].
+    attr_reader :position
+    # Public: Returns the X coordinate of the instance.
+    attr_reader :x
+    # Public: Returns the Y coordinate of the instance.
+    attr_reader :y
+
+    # Public: Initialize the Cell.
+    #
+    # x - The X coordinat
+    # y - The Y coordinate in an X,Y grid.
+    #
+    def initialize(x,y)
+      invalid_type = "Coordinates must be integers!"
+      raise invalid_type unless (x.is_a?(Integer) && y.is_a?(Integer))
+      @position = [x,y]
+      @x = x
+      @y = y
```

Comment:
Even though the Coordinate is essentially an immutable Value Object, and the position won't diverge from x & y unless something goes terribly wrong, I'd still have position be a derived value—just a method rather than an instance variable.